### PR TITLE
Expandable text component

### PR DESCRIPTION
**Pull request by thebailers**
Adds an Expandable Text component within a new typography components directory. The expandable text component takes a title prop, and expects that the content of the element be passed in as children to the component.

### CODE REVIEW

**Comment by JoeChapman** [comment]
I'm going to chime in and say I think this should be two separate PR's or at least two separate commits with the headline being Add Expandable Text Component

File: src/typography/expandable-text.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ExpandableText = ({ title, children }) => (
```
Comment (geemanjs) [inline src/typography/expandable-text.jsx]: I think I added the [`transform-object-rest-spread`](https://babeljs.io/docs/plugins/transform-object-rest-spread/) in a previous PR @lennym could you just validate this all works correctly in the bundled components?

File: src/typography/expandable-text.spec.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import ExpandableText from 'govuk-react-components/typography/expandable-text';
+
+describe('The Expandable Text Component', () => {
+
+  it('should render correctly', () => {
+    expect(
+      render(<ExpandableText title="Lorem ipsum">Dolor set amet</ExpandableText>)
+    ).toMatchSnapshot()
+  });
+
```
Comment (geemanjs) [inline src/typography/expandable-text.spec.js]: Probably worth adding three additional tests here:

1. Passing components in as the 'children' prop
2. Passing components in as the 'title' prop
3. Passing in the `open` prop I mentioned above

File: src/typography/expandable-text.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <summary><span className="summary">{title}</span></summary>
+    <div className="panel panel-border-narrow">
+      {children}
+    </div>
+  </details>
+);
+
+ExpandableText.propTypes = {
+  title: PropTypes.string.isRequired,
+  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired
```
Comment (geemanjs) [inline src/typography/expandable-text.jsx]: I think I missed this on #22 but we should probably also pass in an optional [`open` prop as per the spec](https://dev.w3.org/html5/spec-preview/the-details-element.html)

**Review by geemanjs — APPROVED** [review]
Looks good to me. @lennym are you able to review and merge 👍

**Pull request merged**